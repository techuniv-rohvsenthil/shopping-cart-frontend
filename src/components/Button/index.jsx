import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const Button = (props) => {
  const { children, buttonTestID } = props;
  return (
    <button type="button" className={styles.button} data-testid={buttonTestID}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: propTypes.string.isRequired,
  buttonTestID: propTypes.string.isRequired,
};


export default Button;
