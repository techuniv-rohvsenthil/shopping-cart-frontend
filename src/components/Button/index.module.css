.button{
    background-color: #bdbdbd;
    height: 3rem;
    color: black;
    margin: 0.5rem;
    padding-left: 3rem;
    padding-right: 3rem;
    font-family: 'Poppins';
    border-radius: 10%;
    font-size: 1.1rem;
}

button:active{
    background-color: #f5f5f5;
}
