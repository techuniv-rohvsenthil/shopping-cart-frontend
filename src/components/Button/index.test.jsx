import React from 'react';
import { render } from '@testing-library/react';
import Button from './index';

describe('the Button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button buttonTestID="btn-test">test-data</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
