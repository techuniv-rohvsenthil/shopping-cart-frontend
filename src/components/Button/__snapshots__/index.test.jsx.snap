// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Button component should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="btn-test"
    type="button"
  >
    test-data
  </button>
</DocumentFragment>
`;
